File: code/1. Quick Sort/C++/quick_sort.cpp
#include <iostream>
#include <vector>

void quickSort(std::vector<int> &arr, int low, int high)
{
    if (low < high)
    {
        int pivotIndex = partition(arr, low, high);
        quickSort(arr, low, pivotIndex - 1);
        quickSort(arr, pivotIndex + 1, high);
    }
}

int partition(std::vector<int> &arr, int low, int high)
{
    int pivot = arr[high];
    int i = low - 1;
    for (int j = low; j < high; j++)
    {
        if (arr[j] < pivot)
        {
            i++;
            std::swap(arr[i], arr[j]);
        }
    }
    std::swap(arr[i + 1], arr[high]);
    return i + 1;
}

int main()
{
    std::vector<int> arr = {3, 6, 8, 10, 1, 2, 1};
    quickSort(arr, 0, arr.size() - 1);
    for (int num : arr)
    {
        std::cout << num << " ";
    }
    return 0;
}

File: code/1. Quick Sort/PHP/quick_sort.php
function quickSort(&$arr, $low, $high) {
if ($low < $high) { $pivotIndex=partition($arr, $low, $high); quickSort($arr, $low, $pivotIndex - 1); quickSort($arr,
    $pivotIndex + 1, $high); } } function partition(&$arr, $low, $high) { $pivot=$arr[$high]; $i=$low - 1; for ($j=$low;
    $j < $high; $j++) { if ($arr[$j] < $pivot) { $i++; $temp=$arr[$i]; $arr[$i]=$arr[$j]; $arr[$j]=$temp; } }
    $temp=$arr[$i + 1]; $arr[$i + 1]=$arr[$high]; $arr[$high]=$temp; return $i + 1; } $arr=[3, 6, 8, 10, 1, 2, 1];
    quickSort($arr, 0, count($arr) - 1); foreach ($arr as $num) { echo $num . " " ; }